fix: allows for parent-less package creation (issue #3488).

clemensbartz opened the pull request:
This pull request adds a test case for generating a parent-less package. At the moment, the test case is failing with a NullPointerException on line 39. For this, issue #3488 was raised.

Fix #3488

monperrus commented:
Thanks for the test case, it's very useful.

However, CI does not fail. Any idea?

clemensbartz commented:
Indeed, my test is not run. Which Unit Testing Framework are you using? JUnit 4 or JUnit 5?

MartinWitt commented:
You mixed junit 4 and 5. Spoon supports junit 4 and 5.
Junit 5 assertions come from [here](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html).
Changing the static imports from `org.junit.Assertions` to `org.junit.jupiter.api.Assertions` should fix it.

clemensbartz commented:
When I added the Unit 5 annotations and used the JUnit 5 assertions, the test was still not getting picked up. I switched to Unit 4 and it is failing now because of the issue described in #3488.

monperrus commented:
Thanks a lot for the failing test case. The fix should be relatively easy, would you like to give it a try?

MartinWitt commented:
Fix lgtm, the testcase has 2 issues from my side. 
- It could be moved to a more fitting package and not placed inside the main package 
- in spoon we try to have a description  in natural language as a comment for every testcase following the form: 
`// contract: $discription 
` e.g.
`// contract: a package with null as parentname is creatable `
otherwise good catch and fix

monperrus commented:
@clemensbartz what about moving the test in package `spoon.test.factory`? Thanks!

clemensbartz commented:
Class was moved to package, comment was added as well.

monperrus commented:
Thanks a lot, LGTM. Will merge.

monperrus commented:
Thanks a lot for your contribution @clemensbartz
